### DISCUSSION TITLE
pressing enter key while i'm focuing on one field triggers all the field errors

### DISCUSSION THREAD

**Discussion by YuvarajAnbu** [react-hook-form/react-hook-form#3557] [Q&A] [answered]
i'm creating a sigh up form and the user can click **enter key** to focus on another input, it focuses another input but it triggers validation for all the inputs and shows errors. how do i stop that and trigger validation only on blur or submit?

**Comment by bluebill1049** [comment, marked as answer]
have a look at the doc on the `useForm` section with `mode`

**Reply by YuvarajAnbu** [reply]
i looked at but it doesn't work. i want the form not to submit on pressing enter key

**Reply by bluebill1049** [reply]
then you need to remove `onSubmit` at the form.

**Reply by YuvarajAnbu** [reply]
Ok thx.
